website/server/models/group.js

```diff
@@ -366,6 +366,17 @@ schema.methods.startQuest = async function startQuest (user) {
   let nonUserQuestMembers = _.keys(this.quest.members);
   removeFromArray(nonUserQuestMembers, user._id);
 
+  // remove any users from quest.members who aren't in the party
+  let partyId = this._id;
+  let questMembers = this.quest.members;
+  await Bluebird.map(Object.keys(this.quest.members), async (memberId) => {
+    let member = await User.findOne({_id: memberId, 'party._id': partyId});
```

Comment:
this way the entire user will be fetched. I'd do `User.findOne({_id: memberId, 'party._id': partyId}).select('_id').lean()` so only the _id will be returned and using `.lean()` will prevent a mongoose object from being build instead returning a plain js object which is much faster